src/legacy/core_plugins/input_control_vis/public/control/create_search_source.ts

```diff
@@ -16,16 +16,27 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-import { timefilter } from 'ui/timefilter';
+import { PhraseFilter } from 'src/plugins/data/common/es_query/filters';
+import { IndexPattern, timefilter } from '../legacy_imports';
```

Comment:
Everywhere you're using `IndexPattern` as a type, please use the `IIndexPattern` type from NP instead
```
// Use the IIndexPattern type
import { IIndexPattern } from '../../../../../plugins/data/public';
```

Also, you can use NP timefitler already, even in a legacy plugin:
```
import { npStart } from 'ui/new_platform';
const { timefilter } = npStart.plugins.data.query.timefilter;
```